client/lib/media/store.js

```diff
@@ -31,6 +31,15 @@ function receiveSingle( siteId, item, itemId ) {
 		}
 
 		MediaStore._pointers[ siteId ][ itemId ] = item.ID;
+
+		const previousMediaItem = MediaStore._media[ siteId ][ itemId ];
```

Comment:
> To be honest I don't understand why this code uses item.ID and itemId to perform some operations. Maybe we should rename itemId to previousItemId to make it more clear. Unless it is something else :)

Great catch, I've should have paid more attention to naming here.

If a user uploads a media, [from this code](https://github.com/Automattic/wp-calypso/blob/master/client/lib/media/actions.js#L110-L110) we can see that, firstly, a fake transient media is created and [added to the media library](https://github.com/Automattic/wp-calypso/blob/master/client/lib/media/actions.js#L152-L152). That's the first time `receiveSingle` fn is called (with this transient media). Now, after the [media is uploaded](https://github.com/Automattic/wp-calypso/blob/master/client/lib/media/actions.js#L188-L188), `receiveSingle` is called second time but with the "real", uploaded media. This time, `itemId` is passed as well, containing the transient's media id (for back reference?).

We grab this transient media id (`itemId`) and retrieve the transient media meta (ie description), adding it to uploaded media (which doesn't contain this meta).

Therefore, I'll rename `itemId` to `transientItemId` and `previousMediaItem` to `transientMediaItem` so it becomes a bit more clear.

This might not be the best approach for long term, though. On the other hand, we should (and will) refactor media store into Redux some time so this might be fixed more elegantly in the future.